### PR TITLE
Convert background/context-menu.js into ES Module 

### PR DESCRIPTION
**Pull request by saneyuki**
This pull requests contains these changes:

1. Allow to add es module style file to background module.
2. Enable `no-undef` rule for them.
3. Try to split `gMetricsData` into own module.
4. Convert background/context-menu.js into ES Module.

### CODE REVIEW

**Comment by saneyuki** [comment]
@piroor 

How do you think about this?

File: webextensions/background/module/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import {
+  refreshContextMenuItems,
+  contextMenuClickListener,
```
Comment (piroor) [inline webextensions/background/module/index.js]: There is no need to define these functions as global functions, and I think they should be enclosed under a namespace like: `contextMenu.refreshItems` and `contextMenu.onClick`.

File: webextensions/background/module/metrics.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// Defined in a classic script source, and we can read these as global variables. 
+/* global
+  MetricsData: false,
+ */
+export const gMetricsData = new MetricsData();
```
Comment (piroor) [inline webextensions/background/module/metrics.js]: Why don't you separate this change as another PR? This PR looks doing too much things. Focusing to one topic for each PR, please.

**Comment by piroor** [comment]
Thanks. I've added some comments.

File: webextensions/background/module/index.js (hunk 1 above)
Comment (saneyuki) [reply webextensions/background/module/index.js]: > There is no need to define these functions as global functions, and I think they should be enclosed under a namespace like: contextMenu.refreshItems and contextMenu.onClick.

To not change usages in `background/background.js`, I didn't assign to namespace object.
Should we change them?

File: webextensions/background/module/metrics.js (hunk 2 above)
Comment (saneyuki) [reply webextensions/background/module/metrics.js]: I create https://github.com/piroor/treestyletab/pull/1927/commits/5c737d8eeedd41b49624d12f41e873d5f03c1296 as minimal PoC change to convert es module.

> Focusing to one topic for each PR, please.

Okay. I'll create a new pull request.

**Comment by saneyuki** [comment]
@piroor 

I will close this pull request once, and create pull requests which resolves each of topics.

**Comment by saneyuki** [comment]
I created https://github.com/piroor/treestyletab/pull/1929 as retaking this.

**Pull request closed without merging**